chess_gui.py

```diff
@@ -178,86 +189,58 @@ def main():
         draw_game_state(screen, game_state, valid_moves, square_selected)
 
         endgame = game_state.checkmate_stalemate_checker()
-        if endgame == 0:
-            game_over = True
-            draw_text(screen, "Black wins.")
-        elif endgame == 1:
-            game_over = True
-            draw_text(screen, "White wins.")
-        elif endgame == 2:
-            game_over = True
-            draw_text(screen, "Stalemate.")
+        if endgame in range(3):
+            if endgame == 0:
+                game_over = True
+                draw_text(screen, "Black wins.")
+                logging.info("The game ends with Black's winning.")
+            elif endgame == 1:
+                game_over = True
+                draw_text(screen, "White wins.")
+                logging.info("The game ends with White's winning.")
+            elif endgame == 2:
+                game_over = True
+                draw_text(screen, "Stalemate.")
+                logging.info("The game ends in a stalemate.")
+
+            running = False
+
+            py.display.update()
+            py.time.wait(2000)
+
+            # logs
+            white_check_counter = sum(1 for i in range(0, len(game_state.move_log), 2) if game_state.move_log[i].in_check)
```

Comment:
Generating 1's and summing it up as a counter is creative, but the more accurate way of doing it is to count rather than sum.